crates/edr_provider/src/data/call.rs

```diff
@@ -67,18 +67,20 @@ pub(super) fn run_call(args: RunCallArgs<'_>) -> Result<ResultAndState, Provider
 pub(super) fn run_call_and_handle_errors(
     run_call_args: RunCallArgs<'_>,
     transaction_hash: &B256,
-) -> Result<(u64, Bytes), ProviderError> {
+) -> Result<Result<(u64, Bytes), TransactionFailure>, ProviderError> {
```

Comment:
I'm wondering if we could make `TransactionFailure` a variant of `ProviderError` instead of the nested results to simplify things?

Reply:
I'm trying to figure out how the logic works in EDR and whether that aligns with how Hardhat's implementation works. They have two variables `throwOnCallFailures` and `throwOnTransactionFailures` that determine whether the error is thrown. It seems like we are never doing that in EDR, so this might have to be rewritten.

I'll look into that tomorrow though.